pkg/session/pipelineddmltest/pipelineddml_test.go

```diff
@@ -0,0 +1,392 @@
+	tk.MustExec("set session tidb_mem_quota_query = 128 << 10") // 128KB limitation.
+	tk.MustExec("set session tidb_max_chunk_size = 32")
+	insertStmt := "insert into _t1 select * from t1"
+	err := tk.ExecToErr(insertStmt)
+	require.Error(t, err)
+	require.True(t, strings.Contains(err.Error(), "Your query has been cancelled due to exceeding the allowed memory limit for a single SQL query. Please try narrowing your query scope or increase the tidb_mem_quota_query limit and try again."), err.Error())
+	tk.MustExec("set session tidb_dml_type = bulk")
+	tk.MustExec(insertStmt)
+	tk.MustQuery("select count(*) from _t1").Check(testkit.Rows(fmt.Sprintf("%d", cnt)))
+
+	// update
+	tk.MustExec("set session tidb_mem_quota_query = 256 << 10") // 256KB limitation.
+	updateStmt := "update _t1 set c = 'abcdefghijklmnopqrstuvwxyz1234567890,.?+-=_!@#$&*()_++++++'"
+	tk.MustExec("set session tidb_dml_type = standard")
+	err = tk.ExecToErr(updateStmt)
+	require.Error(t, err)
+	require.True(t, strings.Contains(err.Error(), "Your query has been cancelled due to exceeding the allowed memory limit for a single SQL query. Please try narrowing your query scope or increase the tidb_mem_quota_query limit and try again."), err.Error())
+	tk.MustExec("set session tidb_dml_type = bulk")
+	tk.MustExec(updateStmt)
+	tk.MustQuery("select count(*) from _t1 where c = 'abcdefghijklmnopqrstuvwxyz1234567890,.?+-=_!@#$&*()_++++++'").Check(testkit.Rows(fmt.Sprintf("%d", cnt)))
+
+	// delete
+	tk.MustExec("set session tidb_mem_quota_query = 128 << 10") // 16KB limitation.
+	deleteStmt := "delete from _t1"
+	tk.MustExec("set session tidb_dml_type = standard")
+	err = tk.ExecToErr(deleteStmt)
+	require.Error(t, err)
+	require.True(t, strings.Contains(err.Error(), "Your query has been cancelled due to exceeding the allowed memory limit for a single SQL query. Please try narrowing your query scope or increase the tidb_mem_quota_query limit and try again."), err.Error())
+	tk.MustExec("set session tidb_dml_type = bulk")
+	tk.MustExec(deleteStmt)
+	tk.MustQuery("select count(*) from _t1").Check(testkit.Rows("0"))
+}
+
+func TestPipelinedDMLDisableRetry(t *testing.T) {
+	store := testkit.CreateMockStore(t)
+	tk1 := testkit.NewTestKit(t, store)
+	tk2 := testkit.NewTestKit(t, store)
+	tk1.MustExec("use test")
+	tk2.MustExec("use test")
+
+	tk1.MustExec("drop table if exists t1")
+	tk1.MustExec("create table t1(a int primary key, b int)")
+	tk1.MustExec("insert into t1 values(1, 1)")
+	require.Nil(t, failpoint.Enable("tikvclient/beforePipelinedFlush", `pause`))
+	tk1.MustExec("set session tidb_dml_type = bulk")
+	errCh := make(chan error)
+	go func() {
+		errCh <- tk1.ExecToErr("update t1 set b = b + 20")
+	}()
+	time.Sleep(500 * time.Millisecond)
+	tk2.MustExec("update t1 set b = b + 10")
+	require.Nil(t, failpoint.Disable("tikvclient/beforePipelinedFlush"))
+	err := <-errCh
+	require.Error(t, err)
```

Comment:
How about check the error msg here, I want to known what's kind of error here.

Reply:
Added, it's a write conflict error.